packages/databricks-vscode/src/configuration/AzureCliCheck.ts

```diff
@@ -161,25 +212,44 @@ export class AzureCliCheck implements Disposable {
         return result;
     }
 
-    private async tryLogin(host: URL): Promise<boolean | string> {
+    private async tryLogin(host: URL): Promise<{
+        iss?: string;
+        aud?: string;
+        canLogin: boolean;
+        error?: Error;
+    }> {
         const workspaceClient = new WorkspaceClient({
             host: host.toString(),
             authType: "azure-cli",
+            azureLoginAppId: this.azureLoginAppId,
         });
         try {
             await workspaceClient.currentUser.me();
         } catch (e: any) {
             // parse error message
-            const m = e.message.match(
+            let m = e.message.match(
                 /Expected iss claim to be: https:\/\/sts\.windows\.net\/([a-z0-9-]+?)\/?, but was: https:\/\/sts\.windows\.net\/([a-z0-9-]+)\/?/
             );
             if (m) {
-                return m[1];
+                return {
+                    iss: m[1],
+                    canLogin: false,
+                };
             }
 
-            return false;
+            m = e.message.match(
+                /Expected aud claim to be: ([a-z0-9-]+?)\/?, but was: ([a-z0-9-]+)\/?/
+            );
+            if (m) {
+                return {
+                    aud: m[1],
+                    canLogin: false,
+                };
+            }
+
```

Comment:
AppID can be variable???

Reply:
For staging workspaces there is a different AppID